operational_analysis/methods/quality_check_automation.py

```diff
@@ -0,0 +1,345 @@
+
+        return ws_diurnal
+
+    def wtk_diurnal_plot(self):
+
+        """
+        This method plots the WTK diurnal plot alongisde the hourly power averages of the scada_df across all turbines
+        Args:
+        (None)
+        Returns: (None)
+        """
+
+        if self._id != 'None':
+            scada_sum = self._df.groupby(self._df[self._t])[self._w].sum().to_frame()
+            df_temp = scada_sum.copy()
+        else:
+            df_temp = self._df
+        df_diurnal = df_temp.groupby(df_temp[self._t].dt.hour)[self._w].mean()
+
+        ws_norm = self._ws_diurnal/self._ws_diurnal.mean()
+        df_norm = df_diurnal/df_diurnal.mean()
+    
+        plt.figure(figsize=(8,5))
+        plt.plot(ws_norm, label = 'WTK wind speed (UTC)')
+        plt.plot(df_norm, label = 'QC power')
+        plt.grid()
+        plt.xlabel('Hour of day')
+        plt.ylabel('Normalized values')
+        plt.title('WTK and QC Timezone Comparison')
+        plt.legend()
+        plt.show()
+
+    def corr_df_calc(self):
+        """
+        This method returns a correlation series that compares the current power data (with different shift thresholds) to wind speed data from the WTK with hourly resolution.
+
+        Args:
+        (None)
+
+        Returns:
+        (None)
+        """
+        if self._id != 'None':
+            scada_sum = self._df.groupby(self._df[self._t])[self._w].sum().to_frame()
+            df_diurnal = scada_sum.groupby(scada_sum[self._t].dt.hour)[self._w].mean()
+        else:
+            df_diurnal = self._df.groupby(self._df[self._t].dt.hour)[self._w].mean()
+
+        return_corr = np.empty((24))
+        for i in np.arange(24):
+            df_temp = df_diurnal.shift(i)
+        
+            if i != 0:
+                df_temp[np.arange(i)] = df_diurnal[-i:]
+
+            return_corr[i] = np.corrcoef(self._ws_diurnal['ws'], df_temp)[0,1]
+            
+        self._hour_shift = pd.DataFrame(index = np.arange(24), data = {'Hour Shift Correlation': return_corr})
+
+    def daylight_savings_plot(self, hour_window = 3):
+          
+        """
+        Produce a timeseries plot showing daylight savings events for each year using the passed data.
+
+        Args:
+        hour_window(:obj: 'int'): number of hours outside of the Daylight Savings Time transitions to view in the plot (optional)
+
+        Returns:
+        None
+        """
+    
+        # List of daylight savings days back to 2010
+        dst_df = pd.read_csv('./daylight_savings.csv')
+
+        if self._id != 'None':
+            self._df_dst =  self._df.groupby(self._df[self._t])[self._w].sum().to_frame()
+        else:
+            self._df_dst = self._df
+    
+        self._df_dst['time'] = self._df_dst[self._t]
+        df_full = timeseries.gap_fill_data_frame(self._df_dst, 'time', self._freq) # Gap fill so spring ahead is visible
+        df_full.set_index('time', inplace = True) # Have to reset index to datetime
```

Comment:
Is this needed because timeseries.gap_fill_data_frame is resetting the index? Or are we setting the index for the first time here? If it's the later, what is the meaning of the inline comment?

Reply:
@jordanperr I don't think that time series.gap_fill_data_frame sets the index (but it does sort by time). I have removed the inline comment; it is an artifact from removing the time index dependence for the main DataFrame.

Reply:
@jordanperr Re-evaluating this, I see your point. In addressing other changes, the double index-setting becomes apparent.